examples/sandboxer.rs

```diff
@@ -0,0 +1,141 @@
+use anyhow::bail;
+use landlock::{AccessFs, Ruleset, RulesetAttr};
+use nix::fcntl::{open, OFlag};
+use nix::sys::stat::{fstat, Mode, SFlag};
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+use std::os::unix::process::CommandExt;
+use std::process::Command;
+use std::{
+    borrow::Cow,
+    env,
+    ffi::{OsStr, OsString},
+    path::PathBuf,
+};
+
+const ENV_FS_RO_NAME: &str = "LL_FS_RO";
+const ENV_FS_RW_NAME: &str = "LL_FS_RW";
+
+const ACCESS_FS_ROUGHLY_READ: AccessFs = AccessFs::from_bits_truncate(
+    AccessFs::EXECUTE.bits() | AccessFs::READ_FILE.bits() | AccessFs::READ_DIR.bits(),
+);
+
+const ACCESS_FS_ROUGHLY_WRITE: AccessFs = AccessFs::from_bits_truncate(
+    AccessFs::WRITE_FILE.bits()
+        | AccessFs::REMOVE_DIR.bits()
+        | AccessFs::REMOVE_FILE.bits()
+        | AccessFs::MAKE_CHAR.bits()
+        | AccessFs::MAKE_DIR.bits()
+        | AccessFs::MAKE_REG.bits()
+        | AccessFs::MAKE_SOCK.bits()
+        | AccessFs::MAKE_FIFO.bits()
+        | AccessFs::MAKE_BLOCK.bits()
+        | AccessFs::MAKE_SYM.bits(),
+);
+
+const ACCESS_FILE: AccessFs = AccessFs::from_bits_truncate(
+    AccessFs::READ_FILE.bits() | AccessFs::WRITE_FILE.bits() | AccessFs::EXECUTE.bits(),
+);
+
+fn populate_ruleset(
+    ruleset: Ruleset,
+    paths: OsString,
+    access: AccessFs,
+) -> Result<Ruleset, anyhow::Error> {
+    let mut ruleset = ruleset;
+    let mut actual_access = access.clone();
+    let individual_paths: Vec<PathBuf> = paths
+        .into_vec()
+        .split(|b| *b == b':')
+        .map(|v| OsStr::from_bytes(v).to_owned().into())
+        .collect();
+
+    for path in individual_paths {
+        match open(&path, OFlag::O_PATH | OFlag::O_CLOEXEC, Mode::empty()) {
+            Err(e) => {
+                bail!("Failed to open \"{}\": {}\n", path.to_string_lossy(), e);
+            }
+            Ok(parent_fd) => match fstat(parent_fd) {
+                Ok(stat) => {
+                    if (stat.st_mode & SFlag::S_IFMT.bits()) != SFlag::S_IFDIR.bits() {
+                        actual_access.insert(ACCESS_FILE);
+                    }
+
+                    ruleset = match ruleset.add_path_beneath_rule(parent_fd, actual_access) {
+                        Ok(r) => r,
+                        Err(e) => {
+                            bail!(
+                                "Failed to update ruleset with \"{}\": {}\n",
+                                path.to_string_lossy(),
+                                e
+                            );
+                        }
+                    };
+                }
+                Err(e) => {
+                    bail!("Failed to stat \"{}\": {}\n", path.to_string_lossy(), e);
+                }
+            },
+        }
+    }
+
+    Ok(ruleset)
+}
+
+fn main() -> Result<(), anyhow::Error> {
+    let args: Vec<_> = env::args_os().collect();
+
+    let program_name = args.get(0).map(|s| s.to_string_lossy()).unwrap_or_default();
+
+    if args.len() < 2 {
```

Comment:
This check makes `unwrap_or(Cow::Borrowed("bash -i"))` useless, we still require to set a command.